Library/Homebrew/cmd/search.rb

```diff
@@ -86,7 +86,7 @@ def search
       metacharacters = %w[\\ | ( ) [ ] { } ^ $ * + ? .]
       bad_regex = metacharacters.any? do |char|
         ARGV.any? do |arg|
-          arg.include?(char) && !arg.start_with?("/")
+          arg.include?(char) && !arg.start_with?("/") && !arg.include?("@")
```

Comment:
Yeah, it interprets the `.` in something like `openssl@1.1` as a failed desire to regex.